src/locale/localed.c

```diff
@@ -302,12 +302,22 @@ static int method_set_locale(sd_bus_message *m, void *userdata, sd_bus_error *er
                         if (startswith(*i, name) &&
                             (*i)[k] == '=' &&
                             locale_is_valid((*i) + k + 1)) {
+                                _cleanup_(freelocalep) locale_t loc = 0;
+
                                 valid = true;
 
                                 new_locale[p] = strdup((*i) + k + 1);
                                 if (!new_locale[p])
                                         return -ENOMEM;
 
+                                loc = newlocale(LC_ALL_MASK, new_locale[p], 0);
+                                if (loc == 0) {
+                                        if (errno == ENOMEM)
+                                                return -ENOMEM;
+                                        else
+                                                valid = false;
+                                }
```

Comment:
please move this test to locale_is_valid(), and make that function return "int" instead of "bool" then, i.e. < 0 on ENOMEM and such errors, 0 if invalid, > 0 if valid

Reply:
Now I actually think about it, such change requires pretty big changes elsewhere. The problem is that a bool suddenly turns into a possible -ENOMEM, which, in this case, complicates things needlessly.

There are two things that I can do:
1) Simply ignore the -ENOMEM case in the existing code.
2) Replace all the existing code that uses `locale_is_valid()` to account for -ENOMEM.

I'm in favour of 1.

What do you think?

Reply:
2. The reason is that putting off the pain is not worth it. Experience shows that functions which may fail but don't allow returning a value to show that are always converted to have the return value sooner or later. locale_is_valid() is used in 6 places now, so the refactoring required can't be that bad.

Reply:
yeah, i agree with @keszybz. suppressing error codes never pays off. in the long run we tend to always pay the price for even bigger refactoring if we fall in the trap of being sloppy with it. Please, let's get the error handling right the moment we need it.